### ISSUE TITLE
Can't use the clipboard in other applications - windows 10

### ISSUE THREAD

**Issue by ActarusLeGrand** [darktable-org/darktable#14707]
### Describe the bug

When DT 4.2.1 is in use, after a while the clipboard in windows stop copying from eg: MS one note won't copy any text (Eg text in OneNote itself) the paste button won't work because it won't copy anything. Or MS Edge won't copy anything eather. I close DarkTable and the clipboard start working again. Even if I empty the clipboard, I can't copy any text or image.
I also use the external Editor script, I can't tell if it does come from the script or not.
I nerver had that issue before installing DarkTable. 

### Steps to reproduce

Start Darktable. Work on it, copy and paste from browser to oneNOte. 
It does not start right away. But eventually it won't copy anymore to the clipboard.

### Expected behavior

To copy and paste text and image from any browser to oneNote.

### Where did you install darktable from?

darktable.org

### darktable version

4.2.1

### What OS are you using?

Windows

### What is the version of your OS?

Win 10

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

PNY 1030 2 gb

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

I also use the external Editor script, I can't tell if it does come from the script or not.
I nerver had that issue before installing DarkTable.

**Comment by ActarusLeGrand** [comment]
System
![2023-06-12_16-21-12](https://github.com/darktable-org/darktable/assets/54812718/dce2bd84-1673-4c29-a24e-af1c6b1d498b)

**Comment by ActarusLeGrand** [comment]
In property I also have under Processing,  GPU:  DT ressource to large.

**Comment by wpferguson** [comment]
The external editor script is a lua script that allows you to export an image from darktable and open it in an external editor.  When the editor is closed, the result can be imported back into darktable.

https://github.com/darktable-org/lua-scripts/blob/master/contrib/ext_editor.lua

**Comment by wpferguson** [comment]
I think you might be running out of memory.  You only have 16GB installed of which less that 8GB is available to run darktable.  I would suggest trying default or small for darktable resources.

**Comment by ActarusLeGrand** [comment]
> > I also use the external Editor script, I can't tell if it does come from the script or not.
> 
> What is this external Editor script?

It's the script external editor that come with DT.

**Comment by gi-man** [comment]
> > > I also use the external Editor script, I can't tell if it does come from the script or not.
> > 
> > 
> > What is this external Editor script?
> 
> It's the script external editor that come with DT.

Not really. Lua scripts is something you installed and added to darktable. Running out of memory to hold the image sounds like a likely scenario per Bill's post.

**Comment by wpferguson** [comment]
@ActarusLeGrand when you encounter the clipboard not working start performance monitor and check the memory

**Comment by ActarusLeGrand** [comment]
> @ActarusLeGrand when you encounter the clipboard not working start performance monitor and check the memory

I put the priority to normal and I will monitor.

**Comment by ActarusLeGrand** [comment]
WIth normal, I end up getting the same problem. When I close DT, the clipboard work again. But it's less often. But it just happened just now.

**Comment by ralfbrown** [comment]
@ActarusLeGrand Are you still having this problem with 4.4.1?

**Comment by ActarusLeGrand** [comment]
> @ActarusLeGrand Are you still having this problem with 4.4.1?

Yes.

**Comment by wpferguson** [comment]
Try setting your resources to small and see how that affects it.  When (if) you aren't able to use the clipboard any more, check windows performance monitor and see how much memory you have free.

**Comment by whitenoise-fpv** [comment]
Just wanted to note, that this issue is still in the current master. It hast been for the last years and is really annoying for someone, who also works with the same computer (Imagine being in a collaboration session and suddenly realizing, that you can't copy and paste stuff any more... has been embarrassing more than once).

Here is a discussion in pixls.us
https://discuss.pixls.us/t/darktable-and-the-windows-11-clipboard-bug-or-normal/43077

Stopping DT and restarting the clipboard service can fix it most times.

I have experienced the problem on both of my windows computers. The newer one has pretty good specs and OOM shouldn't be a problem.

**Comment by wpferguson** [comment]
Here's the problem with fixing this

- There are **NO** developers on windows AFAIK
- There isn't a consistent way to reproduce the problem other than keep copying and pasting and sooner or later it will happen.  Even if someone does have a fix they can't be sure because they might not have run it long enough
- It only happens on Windows, so is it a Windows issue or a darktable issue, or a combination of both?

I can reopen this, but until a windows user with dev skills is willing to step up, I don't see this getting fixed.

**Comment by victoryforce** [comment]
> I nerver had that issue before installing DarkTable.

This statement seems to have set the direction for further discussions in finding the cause of the problem in darktable.

But the stated fact is not proof that darktable caused this: https://en.wikipedia.org/wiki/Post_hoc_ergo_propter_hoc

And even if the problem in Windows was really triggered by darktable, this does not mean that there is a bug in darktable. darktable is a program that does not work with the clipboard explicitly. It does not need anything from the clipboard, it does not contain the functionality to export to the clipboard and does not receive anything from there. If GTK implicitly interacts with it somehow (well, what could it do, become a client, a member of clipboard viewer chain?) even in the absence of an explicit request from the program (I highly doubt it, just mentioning it for completeness), then darktable has nothing to do with it, then it might be a GTK bug.

There is one undeniable fact. You cannot raise an issue in the Windows bug tracker. The only thing you can do is post on some Microsoft forum and receive spam responses from "Independent Advisors", who are users just like you who earn extra money by supporting other users. This forces complainers to take the path of least resistance and demand that those who have a bug tracker fix everything. 😀

A Windows service should work correctly regardless of what user programs do. If it doesn't, it's a bug in the OS, not in the programs.

**Comment by gi-man** [comment]
What happens when someone that has this issue pastes the content from the clipboard to `notepad`? What is the content inside the windows clipboard?

**Comment by victoryforce** [comment]
> What happens when someone that has this issue pastes the content from the clipboard to `notepad`? What is the content inside the windows clipboard?

@gi-man As I wrote above, this is irrelevant.

Please see https://learn.microsoft.com/en-us/answers/questions/4103694/clipboard-problems-(can-not-copy-paste)-in-windows

> Imho this happens whatever you do - even nothing - as long as darktable is open long enough.

This observation is very consistent with the conclusion of the author of the bug report at the above link. The issue was not that a specific darktable application was open for a long time, but that the system was idle for a long time, which caused certain background tasks to run that did some bad things.

> To give my opinion I suspect hooks. Please delete hooks. System wide hooks. Pure evil. Especially open source ones written in C for Linux critically acclaimed masterfully incomprehensibly portable. There exists a class of people who just don't read Windows docs.

There is a "class" of people who simply draw far-sighted conclusions without basing themselves on any facts. And they are the ones who start "classifying" other people. If you ask me, that's unacceptable.

In summary, I'm going to close this issue.

**Comment by gi-man** [comment]
That link clearly shows this is a window bug with the clipboard history. It reads that folks discover a workaround. Multiple software seems to cause the history to get full (eg. logi mouse app).

**Comment by ericwj** [comment]
You are just looking for reasons to blame this one someone else, instead of diagnosing, instrumenting and eliminating possible causes and in the meantime looking for reasons to be rude in a witty way.

> Post Hoc, Ergo Propter Hoc

All the information people spent hours of their life obtaining and typing here for the benefit of all is not to be brushed aside by being a smartass like that.

The root cause is Darktable, because this issue is a usability issue. Users report it. Not developers.
* There are other programs that use GTK, most notably GIMP, Inkscape, Audacity, and many more.
* And there are thousands of Win32 programs that I have used extensively for decades now.
* And I treat all those programs exactly like DarkTable; keeping them open even for weeks and hibernating daily instead of rebooting.

Not once have I seen this issue except while using DarkTable.

Hence it is entirely reasonable and intelligent to attribute this issue to DarkTable exclusively and not GTK or for crying out loud Win32.

But yes, your are **entirely** right, **all** that time I **only** experienced the issue **after** installing Darktable **and** installing Darktable was **not** the issue.

That's your **tabula in naufragio**. Go ahead. Close the issue, until the next person running into this reopens it after a week or two of frustration and painstaking elimination of root causes.
